Dockerfile

```diff
@@ -16,32 +21,64 @@ RUN apt-get update && apt-get install -y \
 RUN pip install scikit-learn
 RUN pip install scipy
 
+# download anaconda
+RUN wget https://repo.anaconda.com/archive/Anaconda3-2019.10-Linux-x86_64.sh
+RUN bash Anaconda3-2019.10-Linux-x86_64.sh -b -p /root/anaconda/ && \
+rm Anaconda3-2019.10-Linux-x86_64.sh
+
+# Set path to conda
+ENV PATH=/root/anaconda/bin:$PATH
+
+# Updating Anaconda packages
+RUN conda update conda
+RUN conda update anaconda
+RUN conda update --all
+RUN conda config --append channels conda-forge
+
 # install R dependencies
-RUN CODENAME=`grep CODENAME /etc/lsb-release | cut -c 18-` && \
-    echo "deb http://cran.cnr.berkeley.edu/bin/linux/ubuntu $CODENAME/" >> /etc/apt/sources.list && \
-    sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys E084DAB9 && \
-    sudo apt-get update && \
-    yes '' | sudo apt-get -y install r-base && \
-    yes '' | sudo apt-get -y install r-base-dev && \
-    echo 'install.packages(c("lme4"), repos="http://cran.us.r-project.org", dependencies=TRUE)\n' > /tmp/packages.R && \
-    echo 'install.packages("cluster", dependencies=TRUE, repos="http://cran.us.r-project.org")\n' > /tmp/packages.R && \
-    Rscript /tmp/packages.R
-RUN R -q -e 'if(!require(jsonlite)) install.packages("jsonlite", repos="http://cran.us.r-project.org")' && \
-    R -q -e 'if(!require(clValid)) install.packages("clValid", repos="http://cran.us.r-project.org")' && \
-    R -q -e 'if(!require(amap)) install.packages("amap", repos="http://cran.us.r-project.org")' && \
-    R -q -e 'if(!require(sp)) install.packages("sp", repos="http://cran.us.r-project.org")' && \
-    R -q -e 'if(!require(ape)) install.packages("ape", dependencies=TRUE, repos="http://cran.us.r-project.org")' && \
-    R -q -e 'if(!require(flashClust)) install.packages("flashClust", dependencies=TRUE, repos="http://cran.us.r-project.org")' && \
-    R -q -e 'if(!require(fpc)) install.packages("fpc", dependencies=TRUE, repos="http://cran.us.r-project.org")'
+RUN conda install r
+RUN conda install r-amap 
+RUN conda install r-jsonlite 
+RUN conda install r-clValid 
+RUN conda install r-sp 
+RUN conda install r-ape 
+RUN conda install r-flashClust 
+#RUN conda install r-fpc
+
+RUN R -q -e 'install.packages("fpc", repos="http://cran.r-project.org")'
 
 # -----------------------------------------
 
+
+RUN add-apt-repository ppa:openjdk-r/ppa \
+	&& sudo apt-get update \
+	&& sudo apt-get -y install openjdk-8-jdk \
+	&& echo java versions: \
+	&& java -version \
+	&& javac -version \
+	&& echo $JAVA_HOME \
+	&& ls -l /usr/lib/jvm \
+	&& cd /kb/runtime \
+	&& rm java \
+	&& ln -s /usr/lib/jvm/java-8-openjdk-amd64 java \
+	&& ls -l
+
+ENV JAVA_HOME /usr/lib/jvm/java-8-openjdk-amd64
+
+#RUN cd /kb/dev_container/modules/jars \
+#	&& git pull
+
+# update jars
+RUN cd /kb/dev_container/modules/jars \
+	&& git pull \
+	&& . /kb/dev_container/user-env.sh \
+	&& make deploy
+
 COPY ./ /kb/module
 RUN mkdir -p /kb/module/work
-RUN chmod -R a+rw /kb/module
+RUN chmod 777 /kb/module
```

Comment:
You definitely want the recursive chmod, this is known to cause weird issues if you don't do it

Reply:
All the other dockerfiles use the a+rw syntax, so unless there's a good reason to use 777 here I'd say switch it back